pgmpy/models/MarkovChain.py

```diff
@@ -0,0 +1,299 @@
+        >>> model.add_transition_model('intel', intel_tm)
+        >>> diff_tm = {0: {0: 0.1, 1: 0.5, 2: 0.4}, 1: {0: 0.2, 1: 0.2, 2: 0.6 }, 2: {0: 0.7, 1: 0.15, 2: 0.15}}
+        >>> model.add_transition_model('diff', diff_tm)
+        >>> model.sample(size=5)
+           intel  diff
+        0      0     2
+        1      1     0
+        2      0     1
+        3      1     0
+        4      0     2
+        """
+        # check if the start state is valid
+        if start_state is not None and self._check_state(start_state):
+            self.state = start_state
+        elif start_state is None and self.state is None:
+            raise ValueError('Start state not set.')
+
+        sampled = DataFrame(index=range(size), columns=self.variables)
+        sampled.loc[0] = [self.state[var] for var in self.variables]
+        for i in range(size - 1):
+            for var in self.variables:
+                val = self.state[var]
+                next_val = sample_discrete(list(self.transition_models[var][val].keys()),
+                                           list(self.transition_models[var][val].values()))[0]
+                self.state[var] = next_val
+            sampled.loc[i + 1] = [self.state[var] for var in self.variables]
+        return sampled
+
+    def prob_from_sample(self, state, sample=None, window_size=None):
+        """
+        Given an instantiation (partial or complete) of the variables of the model,
+        compute the probability of observing it over multiple windows in a given sample.
+
+        If 'sample' is not passed as an argument, generate the statisic by sampling from the
+        Markov Chain, starting with a random initial state.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models.MarkovChain import MarkovChain as MC
+        >>> model = MC(['intel', 'diff'], [3, 2])
+        >>> intel_tm = {0: {0: 0.2, 1: 0.4, 2:0.4}, 1: {0: 0, 1: 0.5, 2: 0.5}, 2: {2: 1}}
+        >>> model.add_transition_model('intel', intel_tm)
+        >>> diff_tm = {0: {0: 0.5, 1: 0.5}, 1: {0: 0.25, 1:0.75}}
+        >>> model.add_transition_model('diff', diff_tm)
+        >>> model.prob_from_sample({'diff': 0})
+        array([ 0.27,  0.4 ,  0.18,  0.23, ... 0.29])
+        """
+        if sample is None:
+            # generate sample of size 10000
+            sample = self.sample(self.random_state(), size=10000)
+        if window_size is None:
+            window_size = len(sample) // 100  # default window size is 100
+        windows = len(sample) // window_size
+        probabilities = np.zeros(windows)
+        for i in range(windows):
+            for j in range(window_size):
+                ind = i * 100 + j
+                state_eq = [sample.loc[ind, k] == state[k] for k in state.keys()]
+                if all(state_eq):
+                    probabilities[i] += 1
+        return probabilities / window_size
+
+    def generate_sample(self, start_state=None, size=1):
```

Comment:
This seems to be about 10 faster than the `sample` method.
    %timeit gen = x.generate_sample({'intel': 0, 'diff': 1}, size=10000); [x for x in gen]
    1 loops, best of 3: 515 ms per loop

    %timeit x.sample({'intel': 0, 'diff': 1}, size=10000)
    1 loops, best of 3: 5.7 s per loop